### PR TITLE
feat: add Complex.Kernel

### PR DESCRIPTION
**Pull request by polvalente**

### CODE REVIEW

File: lib/complex/kernel.ex [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    has_pow_operator = function_exported?(Kernel, @pow_atom, 2)
+
+    operators = [+: 2, -: 2, /: 2, *: 2]
+
+    operators =
+      if has_pow_operator do
+        [{@pow_atom, 2} | operators]
+      else
+        operators
+      end
```
Comment (josevalim) [inline lib/complex/kernel.ex]:
```suggestion
    operators = [{@pow_atom, 2}, +: 2, -: 2, /: 2, *: 2]
```

File: lib/complex/kernel.ex [hunk 2]
```diff
@@ -0,0 +1,35 @@
+defmodule Complex.Kernel do
+  @moduledoc """
+  Provides operator overloading for Elixir operators.
+  """
+
+  # This is defined as such so that Elixir 1.12 formatters don't complain
+  # and Elixir 1.13+ formatters don't remove the quotes from :"**" that
+  # would make this work in Elixir <= 1.12
+
+  @pow_atom String.to_atom("**")
```
Comment (josevalim) [inline lib/complex/kernel.ex]:
```suggestion
  # would make this work in Elixir <= 1.12
  @pow_atom String.to_atom("**")

```

File: lib/complex/kernel.ex [hunk 3]
```diff
@@ -0,0 +1,35 @@
+defmodule Complex.Kernel do
+  @moduledoc """
+  Provides operator overloading for Elixir operators.
```
Comment (josevalim) [inline lib/complex/kernel.ex:3]: We need to mention that if you use Complex.Kernel, the arithmetic operators are no longer supported in guards.

File: lib/complex/kernel.ex [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (polvalente) [inline lib/complex/kernel.ex:3]:
```suggestion
  Provides operator overloading for Elixir operators.
  
  When you `use Complex.Kernel`, be aware that the normal operators won't work in clause guards. For that you need to use the fully qualified functions (i.e.: `when Kernel.+(a, b) == 1`) instead.
```

File: lib/complex/kernel.ex (hunk 3 above)
Comment (polvalente) [reply lib/complex/kernel.ex:3]: I've added the paragraph below for that

Comment (polvalente) [reply lib/complex/kernel.ex:3]: I've also updated the README to reflect Complex.Kernel

File: README.md [hunk 5]
```diff
@@ -5,18 +5,19 @@ numbers.
...
+of real and complex numbers.
 
 ## Installation
 
 Add *complex* as a dependency in your `mix.exs` file.
 
 ```elixir
 def deps do
-  [ { :complex, "~> 0.2.0" } ]
+  [{ :complex, "~> 0.3.0" }]
```
Comment (josevalim) [inline README.md]:
```suggestion
  [{:complex, "~> 0.3.0"}]
```

File: lib/complex/kernel.ex [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  When you `use Complex.Kernel`, be aware that the normal operators won't work in clause guards. For that you need to use the fully qualified functions (i.e.: `when Kernel.+(a, b) == 1`) instead.
```
Comment (josevalim) [inline lib/complex/kernel.ex]:
```suggestion
  When you `use Complex.Kernel`, be aware that the arithmetic operators
  won't work in clause guards. For that you need to use the fully qualified
  functions (i.e.: `when Kernel.+(a, b) == 1`) instead.
```

**Review by josevalim — APPROVED** [review]

**Pull request merged**